Make sure the array is initialised or GCC may complain
(seen with Zen4 and GCC 13.2.0)

diff -rupN flux-core-0.74.0.orig/src/common/libutil/basemoji.c flux-core-0.74.0/src/common/libutil/basemoji.c
--- flux-core-0.74.0.orig/src/common/libutil/basemoji.c	2025-05-07 02:46:54.097499500 +0200
+++ flux-core-0.74.0/src/common/libutil/basemoji.c	2025-06-01 15:58:58.708990664 +0200
@@ -147,7 +147,7 @@ int uint64_basemoji_encode (uint64_t id,
 {
     int count;
     int n;
-    char reverse[BASEMOJI_MAXLEN+1];
+    char reverse[BASEMOJI_MAXLEN+1] = {0};
 
     if (buf == NULL || buflen <= 0) {
         errno = EINVAL;
